Refactor and modularize user guide (#826)
pydata/pydata-sphinx-theme: docs/user_guide/layout.rst

        .. button-ref:: layout-article-header
            :color: primary
            :outline:

            Article Header

        **Article Content**

        .. button-ref:: layout-article-footer
            :color: primary
            :outline:

            Article Footer

    .. grid-item::
        :padding: 2
        :outline:
        :columns: 2
        :class: secondary-sidebar

        .. button-ref:: layout-sidebar-secondary
            :color: primary
            :outline:

            Secondary Sidebar

        Within-page header links

    .. grid-item::
        :padding: 2
        :outline:
        :columns: 12
        :class: footer

        .. button-ref:: layout-footer
            :color: primary
            :outline:

            Footer

        Site-wide links.

Horizontal spacing
------------------

By default the theme's three columns have fixed widths.
The ``primary sidebar`` will snap to the left, the ``secondary sidebar`` will snap to the right, and the ``article content`` will be centered in between.

- If one of the sidebars is not present, then the ``article content`` will be centered between the other sidebar and the side of the page.

> - If neither sidebars are present, the ``article content`` will be present in the middle of the page.

Reviewer: ```suggestion
- If neither sidebar is present, the ``article content`` will be in the middle of the page.
```

If you'd like the ``article content`` to take up more width than its default, use the ``max-width`` and ``flex-grow`` CSS variables with the ``.bd-content`` selector.
For example, to make the content grow to fit all available width, add a custom CSS rule like:

.. code-block:: css

   .bd-content {
     flex-grow: 1;
     max-width: 100%;
   }

Templates and components
========================

There are a few major theme sections that you can customize to add/remove
components, or add your own components. Each section is configured with a

> list of *html templates* - these are snippets of HTML that are inserted into

Reviewer: ```suggestion
list of *html templates* — these are snippets of HTML that are inserted into
```

   }

Configure the navbar center alignment
-------------------------------------

By default, the navigation bar center area will align with the content on your
page. This equals the following default configuration:

.. code-block:: python

   html_theme_options = {
      ...
      "navbar_align": "content"
      ...
   }

If instead you'd like these items to snap to the left (closer to the logo), use this
configuration:

.. code-block:: python

   html_theme_options = {
      ...
      "navbar_align": "left"
      ...
   }

If you'd like these items to snap to the right of the page, use this configuration:

.. code-block:: python

   html_theme_options = {
      ...
      "navbar_align": "right"
      ...
   }

.. _layout-article-header:

Article Header
==============

The article header is a narrow bar just above the article's content.
It does not contain anything immediately viewable to the reader, but is kept as a placeholder in case theme developers wish to re-use it in the future.

.. _layout-sidebar-primary:

Primary sidebar (left)
======================

The primary sidebar is just to the left of a page's main content.
It is primarily used for between-section navigation.

> By default it will show links to any sublings / children of the current active top-level section (corresponding to links in your header navigation bar).

Reviewer: ```suggestion
By default it will show links to any siblings / children of the current active top-level section (corresponding to links in your header navigation bar).
```
